### PR TITLE
Fill the diagnostic code field in publish_diagnostics

### LINKED ISSUE #6580: Report diagnostic codes from RA in "problems" pane
Often I'll get spurious errors from RA and I want to disable that specific diagnostic, but it's not easy to find out the name to disable.

I end up having to grep the RA source to find where the message is formatted, which is not ideal. It would be simpler if the diagnostic codes were visible directly, or at least present in the JSON when you copy the error.

### PR DESCRIPTION
**Pull request by Veykril**
Fixes #6580
Before:
Code_znn6VgLLH9
After:
Code_c4jJsvzOEA

### CODE REVIEW

File: crates/rust-analyzer/src/handlers.rs [hunk 1]
```diff
@@ -1127,7 +1127,7 @@ pub(crate) fn publish_diagnostics(
         .map(|d| Diagnostic {
             range: to_proto::range(&line_index, d.range),
             severity: Some(to_proto::diagnostic_severity(d.severity)),
-            code: None,
+            code: d.code.map(|d| d.as_str().to_owned()).map(NumberOrString::String),
```
Comment (kjeremy) [inline crates/rust-analyzer/src/handlers.rs:1130]: In `map_rust_diagnostic_to_lsp()` this is simplified as: `code: code.clone().map(lsp_types::NumberOrString::String)`

**Review by kjeremy — CHANGES_REQUESTED** [review]

File: crates/rust-analyzer/src/handlers.rs (hunk 1 above)
Comment (Veykril) [reply crates/rust-analyzer/src/handlers.rs:1130]: Oh right, I somehow missed that I can do that. Good point 👍

Comment (kjeremy) [reply crates/rust-analyzer/src/handlers.rs:1130]: Makes sense.

**Review by kjeremy — APPROVED** [review]

**Pull request merged**